streaming/src/main/scala/io/gearpump/streaming/executor/Executor.scala

```diff
@@ -179,11 +187,28 @@ class Executor(executorContext: ExecutorContext, userConf : UserConfig, launcher
 
     case query: QueryExecutorConfig =>
       sender ! ExecutorConfig(systemConfig)
+
+    case registered: TaskRegistered =>
+      LOG.debug(s"Register task ${registered.taskId} succeed")
+    case rejected: TaskRejected =>
+      LOG.info(s"Task ${rejected.taskId} is rejected by AppMaster, shutting down myself...")
+      tasks.get(rejected.taskId).foreach(_ ! PoisonPill)
```

Comment:
Should we clean out the reference in field "tasks"?

As executor has watched the task, if task is killed by PoisonPill, executor will restart it automatically.